cpp/src/Slice/Parser.h

```diff
@@ -226,11 +226,9 @@ namespace Slice
         bool hasMetadata(std::string_view directive) const;
         std::optional<std::string> getMetadataArgs(std::string_view directive) const;
 
-        /// Emits a warning unless it should be filtered out by a [["suppress-warning"]].
-        void warning(WarningCategory category, const std::string& file, int line, const std::string& message) const;
+        bool shouldSuppressWarning(WarningCategory category) const;
```

Comment:
I find this function name odd. It probably should be something like `isSuppressed`.

Reply:
Fair enough, that name is simpler.
Changed.